sympy/core/expr.py

```diff
@@ -3352,6 +3352,13 @@ def _eval_derivative(self, s):
             return S.One
         return S.Zero
 
+    def _eval_derivative_n_times(self, s, n):
+        from sympy import Piecewise, Eq
+        if self == s:
+            return Piecewise((self, Eq(n, 0)), (1, Eq(n, 1)), (0, True))
```

Comment:
I think we should use ff here as well. That way the output will be consistent for all monomials.

Reply:
You mean `ff(1, n)*self**(1-n)`? Seems pretty horrible to read.